src/enumerable.cr

```diff
@@ -630,6 +630,12 @@ module Enumerable(T)
   # ```
   # [1, 2, 3, 4, 5].reduce(10) { |acc, i| acc + i } # => 25
   # ```
+  #
+  # The initial value can be of any type
+  #
+  # ```
+  # [1, 1, 2, 3, 2, 3, 1].reduce({} of Int32 => Int32) { |acc, i| acc.merge!({i => 1}){|key, value| value + 1} } # => {1 : 3, 2: 2, 3: 2}
```

Comment:
While having an example is nice, perhaps an example that is a bit easier to understand (and efficient - creating a new hash for each round in the iteration is not how one want to implement that even if it isn't as horrible as before the improvements @asterite did some month ago) might be a good idea here.

Reply:
This example is IMO still too complex. We're not showcasing `merge` here. Why not simply use one of the examples from the spec?

Reply:
Ok, that will be enough